plugins-mu/hm-stack/class-endpoint-controller.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Build endpoint data for HM Stack data on this site.
+ *
+ * @package HMStackIntegration
+ */
+
+namespace HM_Stack;
+
+use WP_REST_Response;
+use WP_Error;
+use WP_Http;
+
+/**
+ * Class Endpoint_Controller
+ */
+class Endpoint_Controller extends \WP_REST_Controller {
+	/**
+	 * Register HM_Stack data routes for the WP REST API.
+	 *
+	 * @todo:: Add schemas to each endpoint.
+	 */
+	public function register_routes() {
+		$namespace = 'hm-stack/v1';
+
+		// Fetch all activity/alerts against this site.
+		register_rest_route( $namespace, 'activity', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_site_activity' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch bandwidth usage for this site.
+		register_rest_route( $namespace, 'bandwidth-usage', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_bandwidth' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch all site environmental data for this site.
+		register_rest_route( $namespace, 'environment-data', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_environment' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+			'schema'              => [
+				'$schema'              => 'http://json-schema.org/draft-04/schema#',
+				'title'                => 'environment-data',
+				'type'                 => 'object',
+				'properties'           => [
+					'environmentData' => [
+						'description'  => esc_html__( 'Data about the current server environment', 'my-textdomain' ),
```

Comment:
I don't think we need i18n, also what's this camelCase?! :P

Reply:
@joehoyle copied over from the `wp_localize_script()` bit in `admin.php`. I tend towards camelcase for any JS variables as it feels more consistent with react and JS itself.

Reply:
I think that makes sense for wp_localize_script, as it's tied specifically to JS executing on that page, but that's not the case for the rest api.